local/src/server/verifier-server.cc

```diff
@@ -440,25 +497,36 @@ TF_Serve(std::thread *t)
     }
 
     // cleanup and get ready for another session.
-    {
-      std::unique_lock<std::mutex> lock(thread_info._mutex);
-      delete thread_info._session;
-      thread_info._session = nullptr;
-    }
+    delete_thread_info_session(thread_info);
   }
 }
 
 void
 TF_Accept(int socket_fd, bool do_tls)
 {
   std::unique_ptr<Session> session;
+  struct pollfd poll_set[1];
+  int numfds = 1;
+  poll_set[0].fd = socket_fd;
+  poll_set[0].events = POLLIN;
+
   while (!Shutdown_Flag) {
     swoc::Errata errata;
+    // Poll so that we can set a timeout and check whether the user requested a shutdown.
+    auto const poll_return = poll(poll_set, numfds, Thread_Sleep_Interval_MS.count());
```

Comment:
Why not `::poll` as you did earlier?

Reply:
The compiler wasn't confused about which to use here. :) ::poll is explicit, so I'll use it.